pkg/loadbalancer/reconciler/termination_test.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package reconciler
+
+import (
+	"context"
+	"iter"
+	"log/slog"
+	"maps"
+	"testing"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/cilium/statedb"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/cilium/cilium/pkg/datapath/sockets"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	lbmaps "github.com/cilium/cilium/pkg/loadbalancer/maps"
+	"github.com/cilium/cilium/pkg/maglev"
+	"github.com/cilium/cilium/pkg/netns"
+	"github.com/cilium/cilium/pkg/testutils"
+)
+
+func TestSocketTermination(t *testing.T) {
+	if testutils.IsPrivileged() {
```

Comment:
Q: I don't see many tests doing this, should this just be something we tool as part of `testutils.Privileged()`?

Reply:
What do you mean? "Tool" as in?

Reply:
If we want to avoid running non-privileged tests on the privileged runner, to save on CI time then we should make that automatically happen, instead of having to manually set this up for each test.

Reply:
Oh you mean I should've just used `testutils.PrivilegedTest()`? Yeah that makes sense. Changed.